game-server/data/handlers/ai/instance/empyreanCrucible/MagePreceptorAI.java

```diff
@@ -56,56 +51,55 @@ public void handleAttack(Creature creature) {
-				break;
-			case 50:
-				SkillEngine.getInstance().getSkill(getOwner(), 19606, 10, getTarget()).useNoAnimationSkill();
-				ThreadPoolManager.getInstance().schedule(() -> {
-					if (!isDead()) {
-						SkillEngine.getInstance().getSkill(getOwner(), 19609, 10, getOwner()).useNoAnimationSkill();
-						ThreadPoolManager.getInstance().schedule(() -> {
-							WorldPosition p = getPosition();
-							spawn(282364, p.getX(), p.getY(), p.getZ(), p.getHeading());
-							spawn(282363, p.getX(), p.getY(), p.getZ(), p.getHeading());
-							scheduleSkill(2000);
-						}, 4500);
-					}
-				}, 3000);
-				break;
-			case 25:
-				SkillEngine.getInstance().getSkill(getOwner(), 19606, 10, getTarget()).useNoAnimationSkill();
-				scheduleSkill(3000);
-				scheduleSkill(9000);
-				scheduleSkill(15000);
-				break;
+			case 75 -> queueSkill(19605, NpcSkillTargetAttribute.RANDOM);
+			case 50 -> handle50PercentPhase();
+			case 25 -> handle25PercentPhase();
 		}
 	}
 
-	private void scheduleSkill(int delay) {
-		ThreadPoolManager.getInstance().schedule(() -> {
+	private void handle50PercentPhase() {
+		queueSkill(19609, NpcSkillTargetAttribute.MOST_HATED);
+		scheduleTask(() -> {
 			if (!isDead()) {
-				SkillEngine.getInstance().getSkill(getOwner(), 19605, 10, getTargetPlayer()).useNoAnimationSkill();
+				queueSkill(19609, NpcSkillTargetAttribute.MOST_HATED);
+				scheduleTask(this::spawnNpcs, 4500);
 			}
-		}, delay);
+		}, 3000);
 	}
 
-	private Player getTargetPlayer() {
-		List<Player> players = new ArrayList<>();
-		getKnownList().forEachPlayer(player -> {
-			if (!player.isDead() && PositionUtil.isInRange(player, getOwner(), 37)) {
-				players.add(player);
-			}
-		});
-		return Rnd.get(players);
+	private void handle25PercentPhase() {
+		queueSkill(19605, NpcSkillTargetAttribute.RANDOM);
+		scheduleRepeatedSkills(3000, 9000, 15000);
+	}
+
+	private void queueSkill(int skillId, NpcSkillTargetAttribute targetAttribute) {
+		getOwner().queueSkill(skillId, 10, 0, targetAttribute);
+	}
+
+	private void scheduleTask(Runnable task, int delay) {
+		ThreadPoolManager.getInstance().schedule(task, delay);
+	}
+
+	private void scheduleRepeatedSkills(int... delays) {
+		for (int delay : delays) {
+			scheduleTask(() -> queueSkill(19605, NpcSkillTargetAttribute.RANDOM), delay);
+		}
+	}
+
+	private void spawnNpcs() {
+		WorldPosition p = getPosition();
+		spawn(282364, p.getX(), p.getY(), p.getZ(), p.getHeading());
+		spawn(282363, p.getX(), p.getY(), p.getZ(), p.getHeading());
+		scheduleTask(() -> queueSkill(19605, NpcSkillTargetAttribute.RANDOM), 2000);
 	}
 
 	private void despawnNpcs() {
-		despawnNpc(getPosition().getWorldMapInstance().getNpc(282364));
-		despawnNpc(getPosition().getWorldMapInstance().getNpc(282363));
+		despawnNpc(282364);
+		despawnNpc(282363);
 	}
 
-	private void despawnNpc(Npc npc) {
+	private void despawnNpc(int npcId) {
+		Npc npc = getPosition().getWorldMapInstance().getNpc(npcId);
```

Comment:
You can simplify it even more and despawn all npc IDs with a single call:
https://github.com/beyond-aion/aion-server/blob/9181add9390d66d67efd21b885ef5eb8d2598403/game-server/data/handlers/ai/instance/nightmareCircus/CircusBoxesAI.java#L71-L75